From f9e8acc6e991f47ebca55a950ad33856c087df19 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 23 Feb 2020 16:08:55 +0200
Subject: [PATCH 08/55] Allow user to add/remove QS with one click

  * from
  https://github.com/LiquidRemix/android_frameworks_base/commit/8b62630afdc8cdee275be256f491bb566c4b30bf

	modified:   packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
---
 .../src/com/android/systemui/qs/customize/TileAdapter.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java b/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
index e738cec4962..c36bc71510b 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
+++ b/packages/SystemUI/src/com/android/systemui/qs/customize/TileAdapter.java
@@ -315,8 +315,13 @@ public class TileAdapter extends RecyclerView.Adapter<Holder> implements TileSta
         holder.mTileView.handleStateChanged(info.state);
         holder.mTileView.setShowAppLabel(position > mEditIndex && !info.isSystem);
 
+        final boolean selectable = mAccessibilityAction == ACTION_NONE || position < mEditIndex;
+        if (!(mAccessibilityManager.isTouchExplorationEnabled() && selectable)) {
+            holder.mTileView.setOnClickListener(v -> move(holder.getAdapterPosition(),
+                    mEditIndex, holder.mTileView));
+        }
+
         if (mAccessibilityManager.isTouchExplorationEnabled()) {
-            final boolean selectable = mAccessibilityAction == ACTION_NONE || position < mEditIndex;
             holder.mTileView.setClickable(selectable);
             holder.mTileView.setFocusable(selectable);
             holder.mTileView.setImportantForAccessibility(selectable
-- 
2.17.1

